src/WebJobs.Extensions.DurableTask/Storage/BlobServiceClientProvider.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Azure.Core;
+using Azure.Storage.Blobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask.Options;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Storage
+{
+    internal sealed class BlobServiceClientProvider : StorageServiceClientProvider<BlobServiceClient, BlobClientOptions, BlobServiceClientProvider.ConnectionOptions>
+    {
+        public BlobServiceClientProvider(IConfigurationSection connectionSection, AzureComponentFactory componentFactory, AzureEventSourceLogForwarder logForwarder)
+            : base(connectionSection, componentFactory, logForwarder)
+        {
+        }
+
+        protected override BlobServiceClient CreateClient(Uri serviceUri, TokenCredential tokenCredential, BlobClientOptions options)
+        {
+            return new BlobServiceClient(serviceUri, tokenCredential, options);
+        }
+
+        internal sealed class ConnectionOptions : StorageServiceConnectionOptions
+        {
+            public Uri BlobServiceUri { get; set; }
+
+            public override Uri ServiceUri => this.BlobServiceUri ?? base.ServiceUri;
```

Comment:
Just a small question, what is the uri for `base.SeviceUri`?  Is it the uri of the storage account?

Reply:
It is the default URI for the Azure Storage Service in the Azure Public Cloud that can be derived from the account name. Each of the services overrides it with their own value, but may default to this one if it was left unspecified explicitly.